### PR TITLE
Weather api

### PR DESCRIPTION
**Pull request by josepbordesjove**

### CODE REVIEW

**Comment by RamonGilabert** [comment]
I'd be careful with the icon since it has an alpha and you'll see the grid.

File: Weather/Weather/MainViewController/Animations/MainController-Animations.swift [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import UIKit
```
Comment (RamonGilabert) [inline Weather/Weather/MainViewController/Animations/MainController-Animations.swift]: Why do you need this class?

File: Weather/Weather/MainViewController/MainController.swift [hunk 2]
```diff
@@ -65,35 +68,76 @@ class MainController: UICollectionViewController, UICollectionViewDelegateFlowLa
...
   override func collectionView(collectionView: UICollectionView, didSelectItemAtIndexPath indexPath: NSIndexPath) {
-    collectionView.deselectItemAtIndexPath(indexPath, animated: true)
+    let cell = collectionView.dequeueReusableCellWithReuseIdentifier(CardsCell.reusableIdentifier,forIndexPath: indexPath) as! CardsCell
+  
+    cell.hourLabel.alpha = 1
     
+    self.collectionView?.layoutSubviews(reloadInputViews())
     
-    print("Something pressed")
+    /*self.collectionView?.reloadItemsAtIndexPaths([indexPath])
```
Comment (RamonGilabert) [inline Weather/Weather/MainViewController/MainController.swift]: If you don't want this you can deleted it.

File: Weather/Weather/MainViewController/Weather API/Time-API.swift [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import UIKit
+
+extension MainController{
```
Comment (RamonGilabert) [inline Weather/Weather/MainViewController/Weather API/Time-API.swift]: Why is this?

File: Weather/Weather/MainViewController/Weather API/Weather-API.swift [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import UIKit
+
+extension MainController{
+  
+  func getWeatherData(){
+    let cityIDs = "\(Card.cardCells[0].cityID),\(Card.cardCells[1].cityID),\(Card.cardCells[2].cityID),\(Card.cardCells[3].cityID),\(Card.cardCells[4].cityID),\(Card.cardCells[5].cityID),\(Card.cardCells[6].cityID),\(Card.cardCells[7].cityID)"
+    
+    let APIkey = "0370938251f23ed7a6f4821164c89029"
+    let requestURL: NSURL = NSURL(string: "http://api.openweathermap.org/data/2.5/group?id=\(cityIDs),1&unitsq=metric&appid=\(APIkey)")!
```
Comment (RamonGilabert) [inline Weather/Weather/MainViewController/Weather API/Weather-API.swift]: Told you to not use `!`.

File: Weather/Weather/MainViewController/Weather API/Weather-API.swift [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    NSURLSession.sharedSession().dataTaskWithURL(requestURL, completionHandler:  { (data: NSData?, response: NSURLResponse?, error: NSError?) in
+        
+        do{
+          let jsonDictionary = try NSJSONSerialization.JSONObjectWithData(data!, options: NSJSONReadingOptions.MutableContainers) as! NSDictionary
```
Comment (RamonGilabert) [inline Weather/Weather/MainViewController/Weather API/Weather-API.swift]: Same thing here.

File: Weather/Weather/MainViewController/Weather API/Weather-API.swift [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+          
+          let list = jsonDictionary["list"] as? NSArray
+          let listLength = jsonDictionary["cnt"] as! Int
```
Comment (RamonGilabert) [inline Weather/Weather/MainViewController/Weather API/Weather-API.swift]: And here.

File: Weather/Weather/MainViewController/Weather API/Weather-API.swift [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+              Card.cardCells[index].minTemperature = round((minTemperature - 273.15)*10)/10
+              Card.cardCells[index].rainChance = clouds
+              Card.cardCells[index].sunriseHour = sunrise
+              Card.cardCells[index].sunsetHour = sunset
+              
+              dispatch_async(dispatch_get_main_queue(), {
+                self.collectionView?.reloadData()
+              })
+            }
+
```
Comment (RamonGilabert) [inline Weather/Weather/MainViewController/Weather API/Weather-API.swift]: You can delete this space.

File: Weather/Weather/MainViewController/Weather API/Weather-API.swift [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+              Card.cardCells[index].sunsetHour = sunset
+              
+              dispatch_async(dispatch_get_main_queue(), {
+                self.collectionView?.reloadData()
+              })
+            }
+            
+          }
+          
+        }catch {
```
Comment (RamonGilabert) [inline Weather/Weather/MainViewController/Weather API/Weather-API.swift]: Space before with the catch.

File: Weather/Weather/MainViewController/Weather API/Time-API.swift (hunk 3 above)
Comment (josepbordesjove) [inline Weather/Weather/MainViewController/Weather API/Time-API.swift]: I can't refresh the collection view, how I do that in the best way?

**Pull request merged**